### DISCUSSION TITLE
Where is the HTML located?

### DISCUSSION THREAD

**Discussion by NSFWHubSocial** [mastodon/mastodon#22087] [Q&A]
I would like to add thumbnail images with hyperlinks for sponsors within the right side navigation panel under "Preferences" but I can't find the HTML anywhere. I've been digging through the .haml files within the "views" directory as well as the other directories but I don't see where the HTML templates are located. Can anyone please point me in the right direction for UI/UX customization?

**Comment by NSFWHubSocial** [comment]
Is the HTML within the database?

**Reply by CiroArtigot** [reply]
Hello, i want to add a link to the blog in the sidebar too.
I find the js that i think renders it
https://github.com/mastodon/mastodon/blob/main/app/javascript/mastodon/features/ui/components/navigation_panel.js
BUt is not working for me, after adding some custom divs to the html, and restarting mastodon services, they not appear

It will be nice to do that :)

**Comment by CiroArtigot** [comment]
I think I have solved the cache issue
These lines in the terminal have solved the problem for me:
`su mastodon`
`RAILS_ENV=production bundle exec rails assets:precompile`
`exit`
`sudo systemctl restart mastodon-*
`
